src/expr/node_manager.h

```diff
@@ -444,7 +444,18 @@ class NodeManager {
   /** Get a Kind from an operator expression */
   static inline Kind operatorToKind(TNode n);
 
-  /** Get corresponding application kindfor function */
+  /** Get corresponding application kind for function
+   *
+   * Different functional nodes are applied differently. For example,
+   * uninterpreted functions are applied via APPLY_UF, while constructors via
+   * APPLY_CONSTRUCTOR. This method provides the correct application according
+   * to which kind of function fun is.
+   *
+   * @param fun The functional node
+   * @return the correct application kind for fun. If fun is not an
```

Comment:
Actually this isn't quite accurate, it just depends on the type. E.g. fun doesn't have to be an uninterpreted function, it just has to have function type?

Reply:
You're right. But then the more fundamental question is: who has "function type"?

Reply:
well the point is that this method is based on type, not term, so just describe it that way.  You actually might even consider having this method take a TypeNode not a Node.

Reply:
I changed the doc. I think it makes sense for this to live in node manager as it refers to "how do I apply this node". Just so happens that it's solely based on its type (which whoever is calling does not necessarily need to know).